bindata/etcd/etcd-restore-backup.sh

```diff
@@ -11,77 +12,50 @@ if [[ $EUID -ne 0 ]]; then
-    exit 1
+function usage {
+  echo 'Path to the directory containing backup files is required: ./etcd-snapshot-restore.sh <path-to-backup>'
+  exit 1
 }
 
-if [ "$1" == "" ]; then
-    usage
-fi
-
-ASSET_DIR=/home/core/assets
-
-RESTORE_STATIC_RESOURCES="true"
-if [ -f "$1" ]; then
-  # For backward-compatibility, we support restoring from single snapshot.db file or single tar.gz file
-  if [[ "$1" =~ \.db$ ]]; then
-    RESTORE_STATIC_RESOURCES="false"
-    SNAPSHOT_FILE="$1"
-  elif [[ "$1" =~ \.tar\.gz$ ]]; then
-    BACKUP_FILE="$1"
-    tar xzf ${BACKUP_FILE} -C ${ASSET_DIR}/tmp/ snapshot.db
-    SNAPSHOT_FILE="${ASSET_DIR}/tmp/snapshot.db"
-  else
-    usage
-  fi
-elif [ -d "$1" ]; then
-  BACKUP_FILE=$(ls -vd "$1"/static_kuberesources*.tar.gz | tail -1) || true
-  SNAPSHOT_FILE=$(ls -vd "$1"/snapshot*.db | tail -1) || true
-  if [ ! -f ${BACKUP_FILE}  -o ! -f ${SNAPSHOT_FILE} ]; then
-    usage
-  fi
-else
+if [ "$1" == "" ] || [ ! -d "$1" ]; then
   usage
 fi
 
-CONFIG_FILE_DIR=/etc/kubernetes
-MANIFEST_DIR="${CONFIG_FILE_DIR}/manifests"
-MANIFEST_STOPPED_DIR="${ASSET_DIR}/manifests-stopped"
-RUN_ENV=/run/etcd/environment
+BACKUP_FILE=$(ls -vd "$1"/static_kuberesources*.tar.gz | tail -1) || true
+SNAPSHOT_FILE=$(ls -vd "$1"/snapshot*.db | tail -1) || true
 
-ETCDCTL="${ASSET_DIR}/bin/etcdctl"
-ETCD_DATA_DIR=/var/lib/etcd
-ETCD_DATA_DIR_BACKUP=/var/lib/etcd-backup
-RESTORE_ETCD_POD_YAML="${CONFIG_FILE_DIR}/static-pod-resources/etcd-certs/configmaps/restore-etcd-pod/pod.yaml"
-ETCD_MANIFEST="${MANIFEST_DIR}/etcd-pod.yaml"
-ETCD_STATIC_RESOURCES="${CONFIG_FILE_DIR}/static-pod-resources/etcd-member"
-STOPPED_STATIC_PODS="${ASSET_DIR}/tmp/stopped-static-pods"
+source etcd.env
 
-source "/usr/local/bin/openshift-recovery-tools"
+if [ ! -f "${SNAPSHOT_FILE}" ]; then
+  echo "etcd snapshot ${SNAPSHOT_FILE} does not exist."
+  exit 1
+fi
 
-function run {
-  init
-  if [ ! -f "${SNAPSHOT_FILE}" ]; then
-    echo "etcd snapshot ${SNAPSHOT_FILE} does not exist."
-    exit 1
-  fi
+# Download etcdctl
+dl_etcdctl
 
-  dl_etcdctl
-  backup_manifest
-  stop_static_pods
-  stop_etcd
-  stop_kubelet
-  stop_all_containers
-  backup_data_dir
-  remove_data_dir
-  [ "${RESTORE_STATIC_RESOURCES}" = "true" ] && remove_kube_static_resources
-  copy_snapshot_to_backupdir
-  [ "${RESTORE_STATIC_RESOURCES}" = "true" ] && restore_kube_static_resources
-  start_static_pods
-  copy_etcd_restore_pod_yaml
-  start_kubelet
-}
+# Move manifests and stop static pods
+if [ ! -d "$MANIFEST_STOPPED_DIR" ]; then
+  mkdir $MANIFEST_STOPPED_DIR
+fi
 
-run
+ls -1d ${MANIFEST_DIR}/* | while read STATIC_POD; do
+  echo "... stopping $STATIC_POD"
+  mv ${MANIFEST_DIR}/${STATIC_POD} $MANIFEST_STOPPED_DIR
+done
+
+# Restore static pod resources
+tar -C ${CONFIG_FILE_DIR} -xzf $BACKUP_FILE static-pod-resources
+
+# Restore snapshot
+${ETCDCTL} snapshot restore $SNAPSHOT_FILE \
```

Comment:
this is embedded in the restore static pod.

Reply:
Yup. Will remove it.